### DISCUSSION TITLE
setReadOnly for RegisteredFileSystemProvider

### DISCUSSION THREAD

**Discussion by bispeira** [CodinGame/monaco-vscode-api#464] [Q&A]
Hello, I have a question about the `RegisteredFile` class, what is the motivation for creating another class called `RegisteredReadOnlyFile` just to control writing/reading permission? Wouldn't it be possible to have this permission control on the `RegisteredFile` with a `setReadOnly()` method so that it would be possible to use a uri a file that change permission depending on external logic?
With that it's possible to use `RegisteredFileSystemProvider` to control file permissions dynamically.

**Comment by CGNonofr** [comment]
I think you are talking about `RegisteredMemoryFile` vs `RegisteredReadOnlyFile`

You can easily make a `RegisteredMemoryFile` readonly, by overriding the `write` method to throw an error

The point of `RegisteredReadOnlyFile` is to be able to provide a loading function, instead of having to provide the content while registering the file (It could probably be better named)
